### PR TITLE
[13.0][mig] fleet_vehicle_stock

### PR DESCRIPTION
**Pull request by iannak**

### CODE REVIEW

**Comment by gurneyalex** [comment]
hello @franzannakarolina  are you still working on this?

**Comment by iannak** [comment]
> hello @franzannakarolina are you still working on this?

yes

**Comment by marcelsavegnago** [comment]
@gurneyalex If you are interested in making this migration, we have no problem with that. For now we are focusing on other subjects and it will probably take some time to focus on this PR

**Review by marcelsavegnago — APPROVED** [review]

**Comment by marcelsavegnago** [comment]
@gurneyalex Migration completed. If you can review I would appreciate it

**Comment by marcelsavegnago** [comment]
ping @ivantodorovich

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2021 - TODAY, Marcel Savegnago <marcel.savegnago@escodoo.com.br>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestFleetVehicle(TransactionCase):
+    def setUp(self):
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
from odoo.tests.common import SavepointCase


class TestFleetVehicle(SavepointCase):
    def setUpClass(cls):
```

nitpicking

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_onchange_product(self):
+        vehicle = self.vehicle
+        vehicle._onchange_product()
+        self.assertFalse(vehicle.current_stock_location_id)
+
+    def test_onchange_model(self):
+        vehicle = self.vehicle
+        vehicle._onchange_model()
+        self.assertTrue(vehicle.product_id == vehicle.model_id.product_id)
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
        self.assertEqual(vehicle.product_id, vehicle.model_id.product_id)
```

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_compute_current_stock_loc_id(self):
+        vehicle = self.vehicle
+        vehicle._compute_current_stock_loc_id()
+        self.assertTrue(vehicle.current_stock_location_id == self.stock_location)
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
        self.assertEqual(vehicle.current_stock_location_id, self.stock_location)
```

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        self.assertFalse(vehicle.current_stock_location_id)
+
+    def test_onchange_model(self):
+        vehicle = self.vehicle
+        vehicle._onchange_model()
+        self.assertTrue(vehicle.product_id == vehicle.model_id.product_id)
+
+    def test_compute_current_stock_loc_id(self):
+        vehicle = self.vehicle
+        vehicle._compute_current_stock_loc_id()
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
```
You shouldn't have to call the compute method directly.
You should rather test that it's recomputed when you want to (e.g.: changing one of the fields in its `api.depends`)

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        self.assertTrue(vehicle.current_stock_location_id == self.stock_location)
+
+        vehicle_without_quant = self.vehicle_without_quant
+        vehicle_without_quant._compute_current_stock_loc_id()
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]: Same comment than before

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        self.assertTrue(vehicle_without_quant.current_stock_location_id.id is False)
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
        self.assertFalse(vehicle_without_quant.current_stock_location_id)
```

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+        vehicle_without_lot_id = self.vehicle_without_lot_id
+        vehicle_without_lot_id._compute_current_stock_loc_id()
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]: Same comment than before

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        self.assertTrue(vehicle_without_lot_id.current_stock_location_id.id is False)
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]:
```suggestion
        self.assertFalse(vehicle_without_lot_id.current_stock_location_id)
```

File: fleet_vehicle_stock/tests/test_fleet_vehicle.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_inverse_fleet_vehicle_model_id(self):
+        product2 = self.product2
+        product2._inverse_fleet_vehicle_model_id()
```
Comment (ivantodorovich) [inline fleet_vehicle_stock/tests/test_fleet_vehicle.py]: You shouldn't call inverse methods directly, but rather test writing on the field that should trigger it.
Here you want to test that the variant's fields are set when you write on the product template's field

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Comment by marcelsavegnago** [comment]
@ivantodorovich Adjustments are done.

**Review by ivantodorovich — APPROVED** [review]
Thanks!

**Pull request merged**